src/components/router.js

```diff
@@ -9,24 +9,20 @@ import DialogFirstLaunch from './dialog/dialog-first-launch'
 const Router = () => {
     const [openFirstLaunch, setFirstLaunch] = useState(true)
     const [isSaved, setSaved] = useState(false)
-    const [isDisabledAuthority, disableAuthority] = useState(false)
+    const [isDisabledAuthority, setAuthority] = useState(false)
+    const { namespace, authority } = useNamespaceDataStore()
 
     useEffect(() => {
-        apiFirstLoad()
-            .then(({ hasNamespace, hasAuthority }) => {
-                if (hasNamespace) {
-                    setFirstLaunch(false)
-                    setSaved(true)
-                } else {
-                    setFirstLaunch(true)
-                    setSaved(false)
-                    disableAuthority(!hasAuthority)
-                }
-            })
-            .catch(e => {
-                console.error(e)
-            })
-    }, [])
+        if (namespace && authority) {
+            setFirstLaunch(false)
+            setSaved(true)
+            setAuthority(!authority)
+        } else if (namespace === false && authority !== undefined) {
+            setFirstLaunch(true)
+            setSaved(false)
+            setAuthority(!authority)
+        }
+    }, [namespace, authority])
```

Comment:
I'm not sure I totally understand what's going on. But it seems like this component could just use the `namespace` value, and the DialogFirstLaunch component could use useDataQuery on its own to get the authorities and set the disabled value.

Reply:
Oh, ok, I'll try that. In case later, I would like to share to other components if a user has `Authority`, should I use a Context?